tmol/score/dunbrack/potentials/dunbrack_pose_score.impl.hh

```diff
@@ -0,0 +1,384 @@
+    TView<Real, 3, D> rotameric_sdev_tables,
+    TView<Vec<int64_t, 2>, 1, D> rotmean_table_sizes,
+    TView<Vec<int64_t, 2>, 1, D> rotmean_table_strides,
+
+    TView<Vec<Real, 2>, 1, D> rotameric_bb_start,        // ntable-set entries
+    TView<Vec<Real, 2>, 1, D> rotameric_bb_step,         // ntable-set entries
+    TView<Vec<Real, 2>, 1, D> rotameric_bb_periodicity,  // ntable-set entries
+
+    TView<Int, 1, D> rotameric_rotind2tableind,
+    TView<Int, 1, D> semirotameric_rotind2tableind,
+
+    TView<Real, 4, D> semirotameric_tables,              // n-semirot-tabset
+    TView<Vec<int64_t, 3>, 1, D> semirot_table_sizes,    // n-semirot-tabset
+    TView<Vec<int64_t, 3>, 1, D> semirot_table_strides,  // n-semirot-tabset
+    TView<Vec<Real, 3>, 1, D> semirot_start,             // n-semirot-tabset
+    TView<Vec<Real, 3>, 1, D> semirot_step,              // n-semirot-tabset
+    TView<Vec<Real, 3>, 1, D> semirot_periodicity,       // n-semirot-tabset
+
+    TView<Int, 1, D> block_n_dihedrals,
+    TView<UnresolvedAtomID<Int>, 2, D> block_phi_uaids,
+    TView<UnresolvedAtomID<Int>, 2, D> block_psi_uaids,
+    TView<UnresolvedAtomID<Int>, 3, D>
+        block_chi_uaids,  // TODO: unused, can probably be removed
+    TView<UnresolvedAtomID<Int>, 3, D> block_dih_uaids,
+    TView<Int, 1, D> block_rotamer_table_set,
+    TView<Int, 1, D> block_rotameric_index,
+    TView<Int, 1, D> block_semirotameric_index,
+    TView<Int, 1, D> block_n_chi,
+    TView<Int, 1, D> block_n_rotameric_chi,
+    TView<Int, 1, D> block_probability_table_offset,
+    TView<Int, 1, D> block_mean_table_offset,
+    TView<Int, 1, D> block_rotamer_index_to_table_index,
+    TView<Int, 1, D> block_semirotameric_tableset_offset,
+
+    bool compute_derivs
+
+    ) -> std::tuple<TPack<Real, 2, D>, TPack<Vec<Real, 3>, 3, D>> {
+  int const n_poses = coords.size(0);
+  int const n_block_types = block_n_dihedrals.size(0);
+  int const max_n_atoms = coords.size(1);
+  int const max_n_blocks = pose_stack_block_coord_offset.size(1);
+  int const max_n_dih = block_dih_uaids.size(1);
+  int const max_n_conns = pose_stack_inter_block_connections.size(2);
+
+  int const DIH_N_ATOMS = 4;  // TODO: is there a global for this?
+
+  assert(coords.size(0) == n_poses);
+  assert(coords.size(1) == max_n_atoms);
+
+  assert(pose_stack_block_coord_offset.size(0) == n_poses);
+  assert(pose_stack_block_coord_offset.size(1) == max_n_blocks);
+
+  assert(pose_stack_block_type.size(0) == n_poses);
+  assert(pose_stack_block_type.size(1) == max_n_blocks);
+
+  assert(pose_stack_inter_block_connections.size(0) == n_poses);
+  assert(pose_stack_inter_block_connections.size(1) == max_n_blocks);
+  assert(pose_stack_inter_block_connections.size(2) == max_n_conns);
+
+  assert(block_type_atom_downstream_of_conn.size(0) == n_block_types);
+  assert(block_type_atom_downstream_of_conn.size(1) == max_n_conns);
+  // assert(block_type_atom_downstream_of_conn.size(2) == ); TODO: what?
```

Comment:
@aleaverfay ?

Reply:
From what I remember from the code, it seems to be the max n atoms of the residues. I don't think there's an easy way to get at that value here.

I tried setting up the asserts to be very verbose, as I think this also serves a dual purpose of 'documenting' the arguments so that it's clear what each dimension is representing.